### PR TITLE
documentation for category package

### LINKED ISSUE #2715: Document java files which require documentation
### Description

The following java files need documentation:

- [ ] Common API Manager.java
- [ ] OpenFoodAPIClient.java
- [ ] WikidataApiClient.java
- [ ] CategoryRepository.java
- [ ] FileDownloader.java
- [ ] CategoryFragmentViewModel.java
- [ ] AdditiveFragmentHelper.java

_Please comment name of few more classes that needs documentation_

### PR DESCRIPTION
**Pull request by vanshikaarora**
## Description

Added Documentation for Category package

## Related issues and discussion
#fixes #2715 and #2717

### CODE REVIEW

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/category/CategoryRepository.java [hunk 1]
```diff
@@ -10,6 +10,10 @@
 import openfoodfacts.github.scrachx.openfood.category.model.Category;
 import openfoodfacts.github.scrachx.openfood.category.network.CategoryNetworkService;
 
+/**
+ * This class recieves list of all categories using CategoryNetworkService
+ * */
+
```
Comment (VaiTon) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/category/CategoryRepository.java]: Remove whitespace

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/category/CategoryRepository.java [hunk 2]
```diff
@@ -21,6 +25,9 @@ public CategoryRepository(CategoryNetworkService networkService, CategoryMapper
         memoryCache = new AtomicReference<>();
     }
 
+    /**
+     * Calling this function retrieves list of all categories from NetworkService*/
+
```
Comment (VaiTon) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/category/CategoryRepository.java]: Format and remove whitespace, add `@return`

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/category/mapper/CategoryMapper.java [hunk 3]
```diff
@@ -10,7 +10,7 @@
 import openfoodfacts.github.scrachx.openfood.category.network.CategoryResponse;
 
 /**
- * Created by Abdelali Eramli on 27/12/2017.
+ * Class used to map tag name with the corresponding categories
  */
```
Comment (VaiTon) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/category/mapper/CategoryMapper.java]: Remove whitespace

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/category/mapper/CategoryMapper.java [hunk 4]
```diff
@@ -19,6 +19,10 @@
     public CategoryMapper() {
     }
 
+    /**
+     * Returns list of Category objects using the tags
+     * @param  tags List of CategoryResponse.Tag object*/
+
```
Comment (VaiTon) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/category/mapper/CategoryMapper.java]: Close comment on next line and remove whitespace

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/category/model/Category.java [hunk 5]
```diff
@@ -1,7 +1,7 @@
 package openfoodfacts.github.scrachx.openfood.category.model;
 
 /**
- * Created by Abdelali Eramli on 27/12/2017.
+ * Model class for Category
  */
```
Comment (VaiTon) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/category/model/Category.java]: Remove whitespace

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/category/network/CategoryResponse.java [hunk 6]
```diff
@@ -6,7 +6,7 @@
 import java.util.List;
 
 /**
- * Created by Abdelali Eramli on 27/12/2017.
+ * Class for response recieved from CaregoryNetworkService class
  */
```
Comment (VaiTon) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/category/network/CategoryResponse.java]: Remove whitespace

**Comment by vanshikaarora** [comment]
@VaiTon i have made some changes. Kindly review :)

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/category/network/CategoryResponse.java (hunk 6 above)
Comment (teolemon) [reply app/src/main/java/openfoodfacts/github/scrachx/openfood/category/network/CategoryResponse.java]: Caregory >> Category

**Comment by teolemon** [comment]
If no update, I propose closing

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/category/CategoryRepository.java (hunk 1 above)
Comment (teolemon) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/category/CategoryRepository.java:16]:
```suggestion
/**
 * This class recieves list of all categories using CategoryNetworkService
 * */
```

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/category/CategoryRepository.java (hunk 2 above)
Comment (teolemon) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/category/CategoryRepository.java:30]:
```suggestion
    /**
     * Calling this function retrieves list of all categories from NetworkService*/
```

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/category/mapper/CategoryMapper.java (hunk 4 above)
Comment (teolemon) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/category/mapper/CategoryMapper.java:25]:
```suggestion
    /**
     * Returns list of Category objects using the tags
     * @param  tags List of CategoryResponse.Tag object
     **/
```

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/category/CategoryRepository.java [hunk 7]
```diff
@@ -9,7 +9,9 @@
 import openfoodfacts.github.scrachx.openfood.category.mapper.CategoryMapper;
 import openfoodfacts.github.scrachx.openfood.category.model.Category;
 import openfoodfacts.github.scrachx.openfood.category.network.CategoryNetworkService;
-
+/**
+ * This class recieves list of all categories using CategoryNetworkService
```
Comment (teolemon) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/category/CategoryRepository.java:13]:
```suggestion
 * This class receives list of all categories using CategoryNetworkService
```

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/category/mapper/CategoryMapper.java [hunk 8]
```diff
@@ -10,15 +10,18 @@
 import openfoodfacts.github.scrachx.openfood.category.network.CategoryResponse;
 
 /**
```
Comment (teolemon) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/category/mapper/CategoryMapper.java:12]:
```suggestion
/**
```

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/category/mapper/CategoryMapper.java [hunk 9]
```diff
@@ -10,15 +10,18 @@
...
- * Created by Abdelali Eramli on 27/12/2017.
+ * Class used to map tag name with the corresponding categories
  */
```
Comment (teolemon) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/category/mapper/CategoryMapper.java:15]:
```suggestion
 */
```

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/category/model/Category.java (hunk 5 above)
Comment (teolemon) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/category/model/Category.java:6]:
```suggestion
 */
```

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/category/model/Category.java [hunk 10]
```diff
@@ -1,7 +1,7 @@
 package openfoodfacts.github.scrachx.openfood.category.model;
 
 /**
```
Comment (teolemon) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/category/model/Category.java:3]:
```suggestion
/**
```

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/category/network/CategoryResponse.java [hunk 11]
```diff
@@ -6,7 +6,7 @@
 import java.util.List;
 
 /**
- * Created by Abdelali Eramli on 27/12/2017.
+ * Class for response recieved from CaregoryNetworkService class
```
Comment (teolemon) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/category/network/CategoryResponse.java:9]:
```suggestion
 * Class for response received from CategoryNetworkService class
```

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/category/mapper/CategoryMapper.java [hunk 12]
```diff
@@ -8,17 +8,19 @@
 
 import openfoodfacts.github.scrachx.openfood.category.model.Category;
 import openfoodfacts.github.scrachx.openfood.category.network.CategoryResponse;
-
 /**
- * Created by Abdelali Eramli on 27/12/2017.
+ * Class used to map tag name with the corresponding categories
  */
```
Comment (teolemon) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/category/mapper/CategoryMapper.java:14]:
```suggestion
 */
```

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/category/model/Category.java [hunk 13]
```diff
@@ -1,7 +1,6 @@
 package openfoodfacts.github.scrachx.openfood.category.model;
-
 /**
- * Created by Abdelali Eramli on 27/12/2017.
+ * Model class for Category
  */
```
Comment (teolemon) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/category/model/Category.java:5]:
```suggestion
 */
```

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/category/network/CategoryResponse.java [hunk 14]
```diff
@@ -6,7 +6,7 @@
 import java.util.List;
 
 /**
- * Created by Abdelali Eramli on 27/12/2017.
+ * Class for response received from CategoryNetworkService class
  */
```
Comment (teolemon) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/category/network/CategoryResponse.java:11]:
```suggestion
 */
```

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/category/network/CategoryResponse.java [hunk 15]
```diff
@@ -6,7 +6,7 @@
 import java.util.List;
 
 /**
```
Comment (teolemon) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/category/network/CategoryResponse.java:8]:
```suggestion
/**
```

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/category/model/Category.java (hunk 13 above)
Comment (teolemon) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/category/model/Category.java:5]:
```suggestion
 */
```

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/category/mapper/CategoryMapper.java (hunk 12 above)
Comment (teolemon) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/category/mapper/CategoryMapper.java:14]:
```suggestion
 * Class used to map tag name with the corresponding categories
 */
```

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/category/model/Category.java (hunk 13 above)
Comment (teolemon) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/category/model/Category.java:5]:
```suggestion
 * Model class for Category
 */
```

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/category/network/CategoryResponse.java [hunk 16]
```diff
@@ -4,9 +4,8 @@
 
 import java.util.Collections;
 import java.util.List;
-
 /**
- * Created by Abdelali Eramli on 27/12/2017.
+ * Class for response received from CategoryNetworkService class
  */
```
Comment (teolemon) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/category/network/CategoryResponse.java:10]:
```suggestion
 * Class for response received from CategoryNetworkService class
 */
```

**Pull request merged**